src/tools.cpp

```diff
@@ -91,6 +92,80 @@ StateHistory sampleSGP4Orbit( const Tle& tle,
     return stateHistory;
 }
 
+//! Convergence test for a virtual TLE generated from the cartesian to TLE converter in atom
+bool virtualTleConvergenceTest( const std::vector< double > propagatedCartesianState,
+                                const std::vector< double > trueCartesianState,
+                                const double relativeTolerance,
+                                const double absoluteTolerance )
+{
+    bool testPassed = false;
+    
+    // check for NAN
+    std::vector< double > absoluteDifference( 6 );
+    for ( int i = 0; i < 6; i++ )
+    {
+        absoluteDifference[ i ] = std::fabs( propagatedCartesianState[ i ] - trueCartesianState[ i ] );
+        bool differenceIsNan = std::isnan( absoluteDifference[ i ] );
+        if ( differenceIsNan == true )
+        {
+            testPassed = false;
+            return testPassed;
+        }
+    }
+
+    // Relative Difference Check
```

Comment:
`// Check if relative error between target and propagated Cartesian states is within specified tolerance.`